crates/net/rpc/src/admin/mod.rs

```diff
@@ -0,0 +1,55 @@
+use jsonrpsee::core::RpcResult;
+use reth_network::{peers::PeerKind, DisconnectReason, NetworkHandle};
+use reth_primitives::NodeRecord;
+use reth_rpc_api::AdminApiServer;
+
+struct AdminApi {
+    /// An interface to interact with the network
+    network: NetworkHandle,
+}
+
+impl AdminApiServer for AdminApi {
+    fn add_peer(&self, record: String) -> RpcResult<bool> {
+        if let Ok(record) = record.parse::<NodeRecord>() {
+            self.network.add_peer(record.id, record.tcp_addr());
+            Ok(true)
+        } else {
+            Ok(false)
+        }
+    }
+
+    fn remove_peer(&self, record: String) -> RpcResult<bool> {
+        if let Ok(record) = record.parse::<NodeRecord>() {
+            self.network
```

Comment:
let's also add a `RemovePeer` command, basically the inverse of `AddPeer`, needs `PeerId, PeerKind`

Reply:
Just to be sure I understood correctly:

- If `RemovePeer` command has `PeerKind::Trusted`, change the peer kind to `PeerKind::Basic`
- If `RemovePeer` command has `PeerKind::Basic` remove the peer from the peerset

Is that right?

Reply:
correct,

this should be the same behavior mentioned for `admin_removeTrustedPeer`:

https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-admin